dataframely/columns/datetime.py

```diff
@@ -265,6 +273,7 @@ def __init__(
         max: dt.datetime | None = None,
         max_exclusive: dt.datetime | None = None,
         resolution: str | None = None,
+        time_zone: ZoneInfo | str | dt.timezone | None = None,
```

Comment:
I'd propose to align the typehint with what `polars` provides, i.e. `str | tzinfo | None`

Reply:
I tried that, but unfortunately `pl.selector.datetime` doesn't accept `tzinfo`, it wants the concrete `timezone` class. This the the typehint that polars provides for the selector: `str | timezone | Collection[str | timezone | None] | None`

Reply:
Additional context: using `timezone` directly seems to be a bit of a footgun since one must provide a static offset and name which may or may not correlate correctly to what the IANA definition is.

Reply:
> I tried that, but unfortunately pl.selector.datetime doesn't accept tzinfo

Given the discussion above, I don't think we need `pl.selector` anymore, so this problem would vanish :)

Reply:
Yep, updated to match `pl.Datetime`